packages/adapter-utils/src/utils.ts

```diff
@@ -180,6 +180,14 @@ export const transformValues = (
       )
       return _.isEmpty(transformed) ? undefined : transformed
     }
+    if (_.isPlainObject(newVal) && !strict) {
+      const transformed = _(newVal)
+        .mapValues((val, key) => transformValue(
+          val,
+          keyPathID?.createNestedID(key),
+        )).omitBy(_.isUndefined).value()
```

Comment:
nit, just to be consistent with the style in the rest of the function
```suggestion
      const transformed = _.omitBy(
        _.mapValues(
          newVal,
          (val, key) => transformValue(val, keyPathID?.createNestedID(key),
        ),
        _.isUndefined,
      )
```